tests/components/cert_expiry/test_config_flow.py

```diff
@@ -3,152 +3,235 @@
 import ssl
 from unittest.mock import patch
 
-import pytest
+from asynctest import patch as asyncpatch
 
 from homeassistant import data_entry_flow
-from homeassistant.components.cert_expiry import config_flow
-from homeassistant.components.cert_expiry.const import DEFAULT_NAME, DEFAULT_PORT
-from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.components.cert_expiry.const import DEFAULT_PORT, DOMAIN
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT, STATE_UNAVAILABLE
 
-from tests.common import MockConfigEntry, mock_coro
+from tests.common import MockConfigEntry
 
-NAME = "Cert Expiry test 1 2 3"
 PORT = 443
 HOST = "example.com"
+GET_CERT_METHOD = "homeassistant.components.cert_expiry.helper.get_cert"
+GET_CERT_TIME_METHOD = (
+    "homeassistant.components.cert_expiry.config_flow.get_cert_time_to_expiry"
+)
+GET_SENSOR_CERT_TIME_METHOD = (
+    "homeassistant.components.cert_expiry.sensor.get_cert_time_to_expiry"
+)
 
 
-@pytest.fixture(name="test_connect")
-def mock_controller():
-    """Mock a successful _prt_in_configuration_exists."""
-    with patch(
-        "homeassistant.components.cert_expiry.config_flow.CertexpiryConfigFlow._test_connection",
-        side_effect=lambda *_: mock_coro(True),
-    ):
-        yield
+async def test_user(hass):
+    """Test user config."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "user"}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
 
+    with asyncpatch(GET_CERT_TIME_METHOD):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={CONF_HOST: HOST, CONF_PORT: PORT}
+        )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == HOST
+    assert result["data"][CONF_HOST] == HOST
+    assert result["data"][CONF_PORT] == PORT
+    assert result["result"].unique_id == f"{HOST}:{PORT}"
 
-def init_config_flow(hass):
-    """Init a configuration flow."""
-    flow = config_flow.CertexpiryConfigFlow()
-    flow.hass = hass
-    return flow
+    with asyncpatch(GET_SENSOR_CERT_TIME_METHOD, return_value=100):
+        await hass.async_block_till_done()
 
+    state = hass.states.get("sensor.cert_expiry_example_com")
```

Comment:
Please separate the sensor tests in a separate sensor test module.

For the config flow tests we just want to test the config flow from every possible starting point to every possible exit point and assert the exit result.

We should patch the `async_setup_entry` in the integration so that we don't set up the sensor after creating the entry.